client/src/components/Savings/AddSavingItem.js

```diff
@@ -0,0 +1,172 @@
+import * as React from 'react';
+import { useDispatch } from 'react-redux';
+import { addSavingsAsync } from '../../redux/thunks/savingThunk';
+
+import { currencies } from '../../resources/transactionOptions.js';
+import {
+    Button,
+    Dialog,
+    DialogActions,
+    DialogContent,
+    DialogContentText,
+    DialogTitle,
+    InputLabel,
+    MenuItem,
+    TextField
+} from '@mui/material';
+
+const default_state = {
+    merchantName: '',
+    amount: '',
+    address: '',
+    date: '',
+    currency: '',
+    description: ''
+}
+
+export default function AddSavingItem() {
+    const [open, setOpen] = React.useState(false);
+    const [openError, setOpenError] = React.useState(false);
+    const [saving, setSaving] = React.useState(default_state);
+    const dispatch = useDispatch();
+
+    const handleChange = (event) => {
+        setSaving({
+            ...saving,
+            [event.target.name]: event.target.value
+        });
+    }
+
+    const resetFields = () => {
+        setOpenError(false);
+        setOpen(false);
+        setSaving(default_state);
+    }
+
+    const closeErrorMessage = () => {
+        setOpenError(false);
+    }
+
+    const handleClickOpen = () => {
+        setOpen(true);
+    };
+
+    const handleAdd = () => {
+        let today = new Date();
+        let savingDate = new Date(saving.date);
+        if (savingDate > today) {
+            setOpenError(true);
+            return;
+        }
+        if (saving.merchantName && saving.amount && saving.address && saving.date && saving.currency) {
+            dispatch(addSavingsAsync([saving]));
+            resetFields();
+        }
+    };
+
+    return (
+        <div>
+            <Button variant="outlined" onClick={handleClickOpen}>
+                Add Saving
+            </Button>
+            <Dialog open={openError} onClose={closeErrorMessage} maxWidth="md">
+                <DialogTitle>Error</DialogTitle>
+                <DialogContent >
+                    <DialogContentText>
+                        Please ensure the saving date is no later than today's date.
+                    </DialogContentText>
+                </DialogContent>
+                <DialogActions>
+                    <Button onClick={closeErrorMessage}>Continue</Button>
+                </DialogActions>
+            </Dialog>
+            <Dialog open={open} onClose={resetFields} maxWidth="md">
+                <DialogTitle>Add Saving</DialogTitle>
+                <DialogContent >
+                    <DialogContentText>
+                        Please enter the saving details below:
+                    </DialogContentText>
+                    <TextField
+                        margin="dense"
+                        id="merchantName"
+                        label="Merchant Name"
```

Comment:
should we maybe change the name for savings? not sure if "merchange name" makes the most sense

Reply:
changed to source!